daphne_worker/src/durable/leader_col_job_queue.rs

```diff
@@ -157,17 +157,14 @@ impl DurableObject for LeaderCollectionJobQueue {
                     ));
                 }
 
+                let mut storage = self.state.storage();
                 let pending_key = format!("pending/{}", collect_id_hex);
-                let pending: Option<OrderedCollectReq> =
-                    state_get(&self.state, &pending_key).await?;
-                if pending.is_none() {
-                    return Err(int_err("LeaderCollectionJobQueue: missing collect request"));
-                }
-
+                let delete_pending_future = storage.delete(&pending_key);
                 self.state
```

Comment:
are you able to reuse the storage variable for the put operation?
also 
`let delete_pending_future = storage.delete(&pending_key).await?;`

Reply:
* Unfortunately no, rustc complains.
* I don't want to await here because I want to be able to run the request below in parallel. That's why the await comes later.